types/null/uint8.go

```diff
@@ -0,0 +1,190 @@
+package null
+
+import (
+	"database/sql/driver"
+	"encoding/json"
+	"fmt"
+	"reflect"
+	"strconv"
+)
+
+// Uint8 is a nullable wrapper around the uint8 type that implementing all of
+// the pyrrho/encoding/types interfaces detailed in the package comments.
+//
+// If the Uint8 is valid and contains 0, it will be considered non-nil, and of
+// zero value.
+type Uint8 struct {
+	Uint8 uint8
+	Valid bool
+}
+
+// Constructors
+
+// NullUint8 constructs and returns a new null Uint8.
+func NullUint8() Uint8 {
+	return Uint8{
+		Uint8: 0,
+		Valid: false,
+	}
+}
+
+// NewUint8 constructs and returns a new, valid Uint8 initialized with the value
+// of the given i.
+func NewUint8(i uint8) Uint8 {
+	return Uint8{
+		Uint8: i,
+		Valid: true,
+	}
+}
+
+// Getters and Setters
+
+// ValueOrZero returns the value of i if it is valid; otherwise it returns the
+// zero value for a uint8 (0).
+func (i Uint8) ValueOrZero() uint8 {
+	if !i.Valid {
+		return 0
+	}
+	return i.Uint8
+}
+
+// Set modifies the value stored in i, and guarantees it is valid.
+func (i *Uint8) Set(v uint8) {
+	i.Uint8 = v
+	i.Valid = true
+}
+
+// Null marks i as null with no meaningful value.
+func (i *Uint8) Null() {
+	i.Uint8 = 0
+	i.Valid = false
+}
+
+// Interfaces
+
+// IsNil implements the pyrrho/encoding IsNiler interface. It will return true
+// if i is null.
+func (i Uint8) IsNil() bool {
+	return !i.Valid
+}
+
+// IsZero implements the pyrrho/encoding IsZeroer interface. It will return true
+// if i is null or if its value is 0.
+func (i Uint8) IsZero() bool {
+	return !i.Valid || i.Uint8 == 0
+}
+
+// Value implements the database/sql/driver Valuer interface. Nil is a valid
+// type to be stored in a driver.Value, but uint8 isn't, so if this Uint8 is
+// valid it will cast its uint8 to an int64.
+func (i Uint8) Value() (driver.Value, error) {
+	if !i.Valid {
+		return nil, nil
+	}
+	return int64(i.Uint8), nil
+}
+
+// Scan implements the database/sql Scanner interface. It will receive a value
+// from an SQL database and assign it to i, so long as the provided data is of
+// type nil, uint8, string, or another integer type that doesn't overflow uint8.
+// All other types will result in an error.
+func (i *Uint8) Scan(src interface{}) error {
+	if i == nil {
+		return fmt.Errorf("null.Uint8: Scan called on nil pointer")
+	}
```

Comment:
I would recommend adding the src-is-null? check here, to mirror `database/sql/sql.go`'s [`NullInt64` implementation](https://golang.org/src/database/sql/sql.go).
```
	if src == nil {
		i.Uint8, n.Valid = 0, false
		return nil
	}
```